### DISCUSSION TITLE
Add Redis to cache queries

### DISCUSSION THREAD

**Discussion by sdomingobasora** [github-stats-22/github-stats#23] [Ideas]
After the last class (23rd November), @manerow @plozanol and I discussed whether we could implement a basic cache using Redis/Memcache. 

The main goal would be for the user to have very quick access to their previous queries (or commands they've executed), including their result even, so that they could save a few roundtrips to the database if this information is requested again. 

However, this feature may not be needed, and may add complexity at some point.

How do you guys feel about this?

**Comment by mikededo** [comment]
As long as this does not extend the complexity for the MVP, I would probably do it. Otherwise, I think it should be something to add once the application has been finished, as an add-on. Basically becasue it is not a functional requirement.

Nonetheless, I'm open to it.

**Reply by plozanol** [reply]
I agree with @mikededo, Redis will be a nice feature once v1.0 is out. If we model a piece that acts as a Proxy, then the Redis addition should be easy. Also, a proxy will help with other incoming data sources like a file for example or db.
